lib/spack/spack/spec.py

```diff
@@ -4793,6 +4793,16 @@ def __init__(self, parse_error):
         self.string = parse_error.string
         self.pos = parse_error.pos
 
+    @property
+    def long_message(self):
+        return "\n".join(
+            [
+                "  Encountered when parsing spec:",
+                "    %s" % self.string,
+                "    %s^" % (" " * self.pos),
```

Comment:
So will this arrow point to a specific spec in the spec string (self.string?)

Reply:
it points to the specific problematic character in the output when it shows the user what it parsed